packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -220,16 +217,28 @@ export class AzureBlobStoragePublish implements PublisherBase {
   ): Promise<TechDocsMetadata> {
     const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
     try {
-      return await new Promise<TechDocsMetadata>(resolve => {
-        const download = this.download(
-          this.containerName,
-          `${entityRootDir}/techdocs_metadata.json`,
-        );
-        resolve(JSON5.parse(download.toString()));
+      return await new Promise<TechDocsMetadata>(async (resolve, reject) => {
```

Comment:
Not quite sure why this was using the `Promise` constructor tbh, prolly better to just get rid of that layer.

Reply:
I didn't quite understand, could you explain in more detail?

Reply:
You should be able to remove this line and the next along with line 235-239 below, then switch the `resolve(techdocsMetadata)`  to `return techdocsMetadata`